Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -14054,6 +14089,354 @@ def pan_os_get_audit_comment_command(args: dict) -> CommandResults:
     )
 
 
+def get_all_profile_names_from_profile_type(profile_type: str, device_group: str) -> list:
+    """
+    Retrieves all profile names from a specified profile type.
+
+    Args:
+        profile_type: The type of profile to retrieve, 'vulnerability' or 'spyware'.
+        device_group: If device_group was provided as a command argument, it will override the configured one.
+
+    Returns:
+        A list of profile names associated with the specified profile type.
+    """
+    
+    #if device_group was provided as a command argument, it will override the configured one
+    if device_group:
+        if device_group == 'shared':
+            xpath= "/config/shared/"
+        else:
+            xpath = "/config/devices/entry/device-group/entry[@name=\'" + device_group + "\']/"
```

Comment:
cant we reuse the code from the existing list command?

Reply:
I am not sure we can because in the existing code it takes the device_code or the vsys from the configuration, but if we get a device_code as a command argument it needs to override the value from the configuration. So thats why i send the device_group as an argument to this function and used this and not the configured one.